File: readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

string sentence;
int count_length(string text);
int count_letters(string text, int length);
int count_words(string text, int length);
int count_sentence(string text, int length);

int main(void)
{
    int length;

    sentence = get_string("Text: ");
    //  printf("%s\n", sentence);
    length = count_length(sentence);
    //  printf("%i letters\n", count_letters(sentence, length));
    //  printf("%i words\n", count_words(sentence, length));
    //  printf("%i sentences\n", count_sentence(sentence, length));
    //  average number of letters over 100 words
    float L = (float) count_letters(sentence, length) / count_words(sentence, length) * 100;
    // printf("%f\n", L);
    //  average number of sentences over 100 words
    float S = (float) count_sentence(sentence, length) / count_words(sentence, length) * 100;
    // printf("%f\n", S);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}

int count_letters(string text, int length)
{
    int count = 0;
    int i;
    for (i = 0; i < length; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text, int length)
{
    int wordcount = 0;
    int lettercount = 0;
    int i;
    for (i = 0; i < length + 1; i++)
    {
        if ((text[i] == 32 || text[i] == 3 || text[i] == 0) && lettercount > 0)
        {
            wordcount++;
            lettercount = 0;
        }
        else if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            lettercount++;
        }
    }
    return wordcount;
}

int count_length(string text)
{
    int length;
    length = strlen(text);
    return length;
}

int count_sentence(string text, int length)
{
    int count = 0;
    int i;
    for (i = 0; i < length + 1; i++)
    {
        if ((text[i] == 33 || text[i] == 46 || text[i] == 63))
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string
from decimal import *
import re


def main():
    sentence = get_string("Text: ").strip()
    length = len(sentence)
    # L = length / (sentence.count(" ") - 1) * 100
    letters = count_letters(sentence)
    # print("letters: " + str(letters))
    words = count_words(sentence)
    # print('words: ' + str(words))
    sentences = count_sentences(sentence)
    # print('sentences: ' + str(sentences))
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    # print(index)
    if (index < 1):
        print("Before Grade 1")
    elif (index > 15):
        print("Grade 16+")
    else:
        print("Grade " + str(round(index)))


def count_letters(sentence):
    p = re.compile('[a-z]', re.I)
    return len(p.findall(sentence))


def count_words(sentence):
    p = re.compile(r'\b[\w\'-]*[^\W]\b')
    wordslist = p.findall(sentence)
    # print(wordslist)
    words = len(wordslist)
    return words


def count_sentences(phrase):
    # p = re.compile(r'[\w\s,]*[^\W]', re.I)
    p = re.compile(r'[\w\s,\'\-\"\:\;]*[^\W]', re.I)
    sentencelist = p.findall(phrase)
    # print(sentencelist)
    sentences = len(sentencelist)
    return sentences


main()
